analysis/ecmwf_switch_zscore.md

```diff
@@ -0,0 +1,563 @@
+```
+
+```python
+df_seas5 = df_seas5_zscore_q.copy()
+```
+
+## SEAS5
+
+### Loading and processing
+
+```python
+# if needed, process SEAS5 rasters (takes a few minutes)
+# seas5.process_seas5_rasters()
+```
+
+```python
+df_seas5 = seas5.load_seas5_stats(variable="zscore")
+```
+
+```python
+# just check the histogram to see that it's sensible
+for issued_month, group in df_seas5.groupby("issued_month"):
+    group["q"].hist(alpha=0.3)
+```
+
+```python
+df_seas5 = calculate_groups_rp(df_seas5, ["issued_month"])
+```
+
+```python
+df_seas5
+```
+
+### Checking combined RP
+
+```python
+df_pivot_rps = df_seas5.pivot(
+    index="year", columns="issued_month", values="q_rp"
+).reset_index()
+df_pivot_rps = df_pivot_rps.rename(columns={x: f"issued_{x}" for x in [3, 7]})
+```
+
+```python
+dicts = []
+min_individual_rp = 2
+
+rp_list = df_seas5["q_rp"].unique()
+rp_list = rp_list[rp_list >= min_individual_rp]
+for rp_3 in rp_list:
+    for rp_7 in rp_list:
+        dff = df_pivot_rps[
+            (df_pivot_rps["issued_3"] >= rp_3)
+            | (df_pivot_rps["issued_7"] >= rp_7)
+        ]
+        dicts.append(
+            {
+                "rp_3": rp_3,
+                "rp_7": rp_7,
+                "rp_overall": (df_seas5["year"].nunique() + 1)
+                / dff["year"].nunique(),
+            }
+        )
+df_rps = pd.DataFrame(dicts)
+```
+
+```python
+heatmap_data = df_rps.pivot(index="rp_7", columns="rp_3", values="rp_overall")
+```
+
+```python
+bounds = [1, 2, 2.5, 3, 4, 5, 6, 10, 1000]
+tick_bounds = bounds[1:-1]
+cmap = plt.cm.Spectral_r
+norm = mcolors.BoundaryNorm(bounds, cmap.N)
+
+fig, ax = plt.subplots(dpi=200, figsize=(8, 8))
+
+sns.heatmap(
+    heatmap_data,
+    annot=False,
+    cmap=cmap,
+    norm=norm,
+    alpha=0.8,
+    cbar_kws={
+        "label": "Période de retour combinée",
+        "ticks": tick_bounds,
+        "shrink": 0.8,
+    },
+    ax=ax,
+)
+ax.invert_yaxis()
+ax.set_aspect("equal", adjustable="box")
+ax.set_title(
+    "Périodes de retour individuelles vs. période de retour combinée,\n"
+    f"prévisions saisonnières SEAS5, depuis {df_seas5['year'].min()}"
+)
+ax.set_xlabel("Mars : période de retour des prévisions")
+ax.set_ylabel("Juillet : période de retour des prévisions")
+
+tick_positions = np.interp(
+    tick_bounds, heatmap_data.columns, range(len(heatmap_data.columns))
+)
+ax.set_xticks(tick_positions)
+ax.set_xticklabels(tick_bounds, rotation=90)
+ax.set_yticks(tick_positions)
+ax.set_yticklabels(tick_bounds)
+
+for x in tick_positions:
+    ax.axvline(x, color="k", alpha=0.5, linewidth=0.5)
+    ax.axhline(x, color="k", alpha=0.5, linewidth=0.5)
+
+plt.show()
+```
+
+### Check trend
+
+```python
+df_pivot = df_seas5.pivot(
+    index="year", columns="issued_month", values="q"
+).reset_index()
+df_pivot = df_pivot.rename(columns={x: f"issued_{x}" for x in [3, 7]})
+```
+
+```python
+df_pivot.plot(x="year", y=["issued_3", "issued_7"])
```

Comment:
Hmmm ok yes looking at this plot again I think 2000 seems like a very reasonable cut off year